PrestaShop/PrestaShop
# Check override conflicts at module install

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch           | 8.2.x
| Description      | This Pull Request introduces new functions to the PrestaShop core that aim to resolve a critical issue related to module overrides. The problem occurs when a module is installed, and it attempts to override a class or method that is already overridden by another module. This can lead to conflicts, unexpected behavior, and errors during the installation process.
| Type             | bug fix
| Category         | BO
| BC breaks        | no
| Deprecations     | no
| How to test      | see below
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/10918924024
| Fixed issue or discussion     | https://github.com/PrestaShop/PrestaShop/issues/36617
| Related PRs       | 
| Sponsor company   | Prestashop SA

# HOW TO TEST

- Install module pscsx3241, installation should be succesful
- Install module pscsx32412, installation should fail
- check in db, in table `ps_module` that you have  pscsx3241
- check in db, in table `ps_module` that you *don't* have module pscsx32412

 
[pscsx3241.zip](https://github.com/user-attachments/files/17025420/pscsx3241.zip)
[pscsx32412.zip](https://github.com/user-attachments/files/17025421/pscsx32412.zip)
